Version 1

mschottdorf opened the pull request:
PR to transfer all Solidworks assembly and part files to this repository. Also added:

1. Big picture hardware design history in a README file.
2. Assembly Instructions with pictures in a pdf.
3. Minor changes in the documentation to point to local files

mschottdorf committed:
Version 1

mschottdorf committed:
updates to docs with links to SolidWorks / pdf instructions

mschottdorf committed:
updated BOM / docs

mschottdorf committed:
some cleanup

mschottdorf committed:
added OKH-manifest

mschottdorf committed:
Update to hardware readme.md

sneakers-the-rat commented:
the nondeterministic test behavior looks like it's coming from here, just not throwing a connectionrefused error. since it's just a pass (i think that's fine? since it's in the close method?) would it be ok to exclude from coverage? or else is it important that we figure out what would cause that exception?

mschottdorf committed:
Adding a "no cover" to coordinator's "stop" method

... as this just runs into a `...pass`

mschottdorf commented:
Fair enough. But it is odd, that pytest has issues with `connectionrefused`. Maybe it is a timeout issue in the VR running the tests itself.

sneakers-the-rat committed:
fixing docs
- limit pyqtgraph version with ~= instead of >= in requirements_docs.txt
- don't include extra copies of assets from built docs (will remove in following PR)

sneakers-the-rat committed:
setuptools packaging hurting my soul

sneakers-the-rat committed:
[skip travis] debugging docs and internal links

sneakers-the-rat committed:
[skip travis] Splitting BOM table

sneakers-the-rat committed:
[skip travis] PySide2<5.14 not compatible with python >=3.8

sneakers-the-rat committed:
[skip travis] does the download role let us link to these files?

sneakers-the-rat committed:
[skip travis] Fixing links and table widths

sneakers-the-rat reviewed (commented):
The changes in the PR are fine, but since our docs aren't building automatically I don't think you would have been able to check them out yet. I have just been manually building them ( https://www.peoplesvent.org/en/tmp_hardware/index.html ), but there are a number of problems with broken links and other formatting problems. I think you should probably go through the pages on the docs and make sure they look how you want to look/all the links that should be there are there. 

I  also needed to fix some problems in the required versions, should just affect the docs, but also there were some bugs in the controller docstring formatting that were throwing some worrying looking errors in the sphinx build that i tried to smooth out. Didn't change any code there.

sneakers-the-rat commented on _docs/hardware/assembly.md:
```diff
@@ -10,10 +10,9 @@ in the **Bill of Materials**:
 
-The entire Solidworks Assembly, with associated part files, can be found in this Google Drive: 
-https://drive.google.com/drive/folders/1YrJEOmOMZtXcHABYO0hEu0etJ7LAC9Lf?usp=sharing
+The entire Solidworks Assembly, with associated part files, can be found [here](/assets/solidworks/PVP1_Mk3.SLDASM), together with two earlier versions of the device, [Mk1](/assets/solidworks/PVP1_Mk1.SLDASM) and [Mk2](/assets/solidworks/PVP1_Mk2.SLDASM). All associated parts are collected [here](/assets/solidworks/Parts).
```
Made a list version with :download: directives, testing now

sneakers-the-rat commented on _docs/hardware/bom.rst:
```diff
@@ -1,8 +1,11 @@
+
 .. csv-table:: PVP bill of materials
-   :file: /assets/csv/10_27_2020_bom.csv
+   :file: /assets/csv/10_27_2020_bom_short.csv
```
Split this up into separate tables so it's more browsable/can navigate with the TOC. May want to double check my edits?

mschottdorf replied:
The separate tables look alright to me. If anyone is interested in the details, they'll have to download the full CSVs anyway 👍

sneakers-the-rat committed:
Fourth column breaks setting explicit widths, but removing that for now

mschottdorf committed:
this method is tested above

mschottdorf commented on _docs/assets/csv/10_27_2020_bom_short.csv:
```diff
@@ -0,0 +1,110 @@
+Component,Subtotal (Bulk Parts Cost) ,Example Sourcing Link,
```
yeah, that is the list from @juliennelachance. No changes, just reformatting

mschottdorf commented on _docs/hardware/assembly.md:
```diff
@@ -10,10 +10,9 @@ in the **Bill of Materials**:
+The entire Solidworks Assembly, with associated part files, can be found [here](/assets/solidworks/PVP1_Mk3.SLDASM), together with two earlier versions of the device, [Mk1](/assets/solidworks/PVP1_Mk1.SLDASM) and [Mk2](/assets/solidworks/PVP1_Mk2.SLDASM). All associated parts are collected [here](/assets/solidworks/Parts).
 
-The current assembly is named `TLA_VENTILATOR_ASSY_V2.SLDASM`.
+The current assembly is named `PVP1_Mk3.SLDASM`. We also provide more detailed build instructions in this [pdf file](/assets/PVP1 Assembly Instructions.pdf).
```
yes, it is exactly the same, just in one document. Phrasing this as "PDF version of the instructions" sounds good to me.

mschottdorf commented:
Thanks for going through this all @sneakers-the-rat! Again, BOM looks alright to me. Definitely easier to read. Links to solidwork drawings work, too, and I like the "prior version" list on the assembly page. 

I'll send a note to Daniel Suo to ask if they want to add their designs in a new PR

mschottdorf commented:
The PR includes:
- All solid work parts, and multiple earlier versions of PVP1.
- More documentation, such as split BOMs, and assembly instructions as pdf.
- Adding the `okh-manifest.yml` file.
- Minor changes to travis and removing redundant checks.
